install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml

```diff
@@ -223,6 +223,15 @@ spec:
 {{- if .Values.nodeinit.restartPods }}
               echo "Restarting kubenet managed pods"
               if [ ! -f /etc/crictl.yaml ] || grep -q 'docker' /etc/crictl.yaml; then
+                # Wait until cilium-agent is up and running which means the cni
+                # configuration and cni plugin are deployed.
+                echo "Waiting for Cilium agent to be up and running"
+                until docker ps | grep cilium-agent | grep -q Running; do echo -n "." && sleep 1; done
```

Comment:
I'm curious what kind of testing we've done / could put in place for this change?

Seems like it should work since this DS runs with `HostPID: true`, though I don't have a local reference what the `docker ps` output is expected to look like in these cases and whether there's any limitations there. I guess it'll be pretty obvious if this gets stuck forever, do we have similar `docker ps` handling logic in this script already?

Reply:
If I understood your question correctly, yes line 90 in this file. https://github.com/cilium/cilium/pull/16588/files#diff-73e3d217d30622143e06f8c117c577385c0cde6144c53853c6b49c9df57c58acL90